WordImpress/Give-Addon-Boilerplate
# Feature: add sample integration for off-site gateways

## Description



This PR adds a Sample Off-Site Gateway Integration which can be used as a starting point (and documentation) for new off-site gateway integrations.

## Visuals



**Demo Video:** 
https://www.loom.com/share/12f3cf0338494e03bc9a3170c064b2f1?sid=9cfd3437-aaac-4cfc-9147-ba6362dc3594

**Off-site Checkout Page Simulation:**

![image](https://github.com/user-attachments/assets/8e0794cd-5ca0-4115-9544-82f7c3c80881)

## Testing Instructions



1. Pull this branch and then follow these steps: https://github.com/impress-org/givewp-addon-boilerplate?tab=readme-ov-file#setup--installation
2. Go to the `Give > Settings > Payment Gateways > Gateways` page and enable the sample off-site gateway for both: **Visual Form Builder** and **Option-Based Form Editor**;
3. Create a few test donations and make sure everything works as expected.


## Pre-review Checklist
